analyses/independent-samples/00-repeated-samples.Rmd

```diff
@@ -0,0 +1,223 @@
+    toc_float: true
+---
+
+## Purpose
+
+There are many specimens in the full dataset that are repeated samples from the same participants.
+This workbook is a quick summary of those samples, so that we can better select a unique set of specimens for downstream analysis. 
+This is particularly important for analyses such as mutation co-occurence, where repeated sampling will inflate apparent co-ocurrence measures.
+
+
+```{r setup}
+# load libraries
+library(dplyr)
+
+# source sample selection funciton
+source("independent-samples.R")
+
+set.seed(201910)
+```
+
+
+```{r load files}
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+sample_df <- readr::read_tsv(file.path(root_dir, "data", "pbta-histologies.tsv"))
+```
+
+## WGS and WXS samples
+Sort to just sequencing samples of tumors
+```{r}
+seq_samples <- sample_df %>%
+  filter(experimental_strategy %in% c("WGS", "WXS"), 
+         sample_type == "Tumor")
+```
+
+How many WGS, how many WXS, summarize #s of samples and participants by 
+```{r}
+seq_samples %>%
+  group_by(experimental_strategy, composition, tumor_descriptor) %>%
+  summarise(samples = length(unique(Kids_First_Biospecimen_ID)),
+            participants = length(unique(Kids_First_Participant_ID)), 
+  ) 
+```
+
+What if we reduce to the earliest sample for each participant?
+
+```{r}
+early_samples <- seq_samples %>%
+  group_by(Kids_First_Participant_ID) %>%
+  summarize(age_at_diagnosis_days = min(age_at_diagnosis_days)) %>%
+  left_join(seq_samples) %>%
+  pull(Kids_First_Biospecimen_ID) %>%
+  unique()
+
+seq_samples %>% 
+  filter(Kids_First_Biospecimen_ID %in% early_samples) %>%
+  group_by(experimental_strategy, composition, tumor_descriptor) %>%
+  summarise(samples = length(unique(Kids_First_Biospecimen_ID)),
+            participants = length(unique(Kids_First_Participant_ID)), 
+  )
+```
+
+There are still many samples for which there are more samples than participants, indicating duplicates, even among the earliest samples.
+
+
+## Derived cell lines
+
+Are there any samples for which we only have derived cell lines?
+
+```{r}
+derived_participants <- seq_samples %>%
+  filter(composition == "Derived Cell Line") %>%
+  pull(Kids_First_Participant_ID) %>%
+  unique() 
+
+non_derived_participants <- seq_samples %>% 
+  filter(composition != "Derived Cell Line") %>%
+  pull(Kids_First_Participant_ID) %>%
+  unique() 
+
+notissue_participants  <- derived_participants[!(derived_participants %in% non_derived_participants)]
+length(notissue_participants)
+
+```
+
+Looks like there is a cell line sample with no corresponding tumor.
+
+```{r}
+seq_samples %>%
+  filter(Kids_First_Participant_ID %in% notissue_participants)
+```
+
+Since this is only one sample it seems like the best move will be to remove all derived cell lines.
+
+## Unlabeled composition samples
+
+9 samples are listed as "Not Reported" for `composition`. 
+What are these, and do we have correponding samples for which composition is known?
```

Comment:
Update: I will change all of the Tumor Not Reported to Tumor Solid Tissue as I confirmed these are not cell lines. However, we cannot change the Normals since this information did not exist at sample collection. However, if it was really imperative, we could probably do GSEA for tissue type and predict :P! Since some composition data is missing, we can use `sample_type` once removing cell lines, from which to select our dataset.

Reply:
For this analysis, I am already looking at only tumors, so that should clear everything up. Thanks for looking into it!